cmd/apiregister-gen/generators/versioned_generator.go

```diff
@@ -57,14 +57,26 @@ func (d *versionedGenerator) Finalize(context *generator.Context, w io.Writer) e
 var VersionedAPITemplate = `
 var (
 	{{ range $api := .Resources -}}
-	{{$api.Group}}{{$api.Kind}}Storage = builders.NewApiResource( //  Resource endpoint
-			{{ $api.Group }}.Internal{{ $api.Kind }},
+
+	{{if $api.REST }}
+		{{$api.Group}}{{$api.Kind}}Storage = builders.NewApiResourceWithStorage( // Resource status endpoint
+			{{ $api.Group }}.Internal{{ $api.Kind }}Status,
```

Comment:
What is the usecase for this?  Why are we continuing to create the status endpoint in this case?